feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/PostActions.kt

```diff
@@ -46,24 +45,24 @@ fun rememberPostActions(
         onRequestRetrySend,
         clipboardManager
     ) {
-        if (post != null) {
-            val doesPostExistOnServer = post.serverPostId != null
-            val hasEditPostRights = hasModerationRights || post.authorId == clientId
-            val hasResolvePostRights = isAtLeastTutorInCourse || post.authorId == clientId
-
-            PostActions(
-                requestEditPost = if (doesPostExistOnServer && hasEditPostRights) onRequestEdit else null,
-                requestDeletePost = if (hasEditPostRights) onRequestDelete else null,
-                onClickReaction = if (doesPostExistOnServer) onClickReaction else null,
-                onCopyText = {
-                    clipboardManager.setText(AnnotatedString(post.content.orEmpty()))
-                },
-                onReplyInThread = if (doesPostExistOnServer) onReplyInThread else null,
-                onResolvePost = if (hasResolvePostRights) onResolvePost else null,
-                onRequestRetrySend = onRequestRetrySend
-            )
-        } else {
-            PostActions()
+        if (post == null) {
+            return@remember PostActions()
         }
+
+        val doesPostExistOnServer = post.serverPostId != null
+        val hasEditPostRights = post.authorId == clientId
+        val hasResolvePostRights = isAtLeastTutorInCourse || post.authorId == clientId
+
+        PostActions(
+            requestEditPost = if (doesPostExistOnServer && hasEditPostRights) onRequestEdit else null,
+            requestDeletePost = if (hasEditPostRights) onRequestDelete else null,
```

Comment:
`hasEditPostRights` is used for deleting, too. However, deleting should be possible for moderators. Could you adjust this?

Reply:
Very true, good catch!